### ISSUE TITLE
wallet.accounts throwing error

### ISSUE THREAD

**Issue by ckgagan** [ruby-wallet/ruby-wallet#1]
I tried the following

 require 'bitcoin'
wallet = RubyWallet.connect(:username => "username", :password => "password", :host => "host", :port => 8332, :ssl => false)
wallet.balance gives me 0.0 which is as expected

but

wallet.accounts throws error

2.1.1 :013 > wallet.accounts
NoMethodError: private method `client' called for #<RubyWallet::Wallet:0x00000002c45fc0>
    from /home/ckgagan/.rvm/gems/ruby-2.1.1/gems/ruby-wallet-0.0.13/lib/wallet/accounts.rb:5:in`client'
    from /home/ckgagan/.rvm/gems/ruby-2.1.1/gems/ruby-wallet-0.0.13/lib/wallet/accounts.rb:13:in `initialize'
    from /home/ckgagan/.rvm/gems/ruby-2.1.1/gems/ruby-wallet-0.0.13/lib/wallet/wallet.rb:21:in`new'
    from /home/ckgagan/.rvm/gems/ruby-2.1.1/gems/ruby-wallet-0.0.13/lib/wallet/wallet.rb:21:in `accounts'
    from (irb):13
    from /home/ckgagan/.rvm/rubies/ruby-2.1.1/bin/irb:11:in`<main>'

**Comment by ruby-wallet** [comment]
I'm moving this over to my other account to continue development. I will look into this issue.

**Comment by drunkonsound** [comment]
Okay, I fixed that issue. I combined two projects and one was not written very well but the concept was pretty good. I have been slowly working out the bugs. If you notice anything else please let me know I will be happy to resolve it.

**Comment by ogminor** [comment]
The fix above was not enough, after auditing the code. I realized that the previous developer had been abusing delegate to expose a private function client. This function should not have been exposed and was a big security risk. Instead I abstracted all the functionality needed into generalized functions in wallet. This makes more logical sense, protects sensitive functionality and overall is better. 

I also removed the ability to dump private keys. There are ways this can be useful but for my purposes it only adds more danger.
